packages/devtools_app/lib/src/memory/memory_controller.dart

```diff
@@ -145,22 +154,39 @@ class MemoryController extends DisposableController
   // List of completed Analysis of Snapshots.
   final List<AnalysisSnapshotReference> completedAnalyses = [];
 
-  bool enableAnalyzeButton() {
-    if (snapshots.isNotEmpty) {
-      final lastSnapshot = snapshots.last;
-      if (completedAnalyses.isNotEmpty) {
-        final result = completedAnalyses.last.dateTime.compareTo(
-              lastSnapshot.collectedTimestamp,
-            ) !=
-            0;
-        return result;
-      }
-      return true;
+  /// Compute the snapshot to analyze.
+  Snapshot getSnapshotToAnalyze() {
```

Comment:
avoid prefixing method names with the word get. `computeSnapshotToAnalyze` or just `get snapshotToAnalyze` if the performance is good enough.

Reply:
get computeSnapshotToAnalyze